### PR TITLE
Bugfix: Correct Google Analytics usage

### PR DESCRIPTION
**Pull request by joaomagfreitas**
Previously in March of 2021, @djatwood provided support for Google Analytics v4 API (newest version of Google Analytics) in Hugo official default templates ([hugo#8287](https://github.com/gohugoio/hugo/pull/8287)), allowing users that use such version to track their website. Although the introduction of this change, the theme had his own version of Google Analytics, which only supports the v3 API and thus was ignoring the v4 change.

This PR fixes that, by removing the outdated implementation from the theme template partials and referencing the official implementation in the `head.html` partial template. v3 also still supported.

## References

- https://github.com/gohugoio/hugo/pull/8287
- https://gist.github.com/zjeaton/42246742cdaf2fb46400d04c2eba9a8a#gistcomment-3646274
- https://gohugo.io/templates/internal/#google-analytics

### CODE REVIEW

File: layouts/partials/head.html [hunk 1]
```diff
@@ -29,10 +29,14 @@
   {{ $customCssPath := "css/custom.css" }}
   {{ $customCss:= resources.Get $customCssPath | resources.ExecuteAsTemplate $customCssPath . }}
   <link rel="stylesheet" type="text/css" href="{{ $customCss.RelPermalink }}">
+  
+  {{ if and (hugo.IsProduction) }}
```
Comment (zwbetz-gh) [inline layouts/partials/head.html]: No need for and here

```suggestion
  {{ if hugo.IsProduction }}
```

**Review by zwbetz-gh — COMMENTED** [review]
Thanks for this. 

One minor comment.

**Comment by joaomagfreitas** [comment]
> Thanks for this.
> 
> One minor comment.

Thanks for the update!

**Comment by zwbetz-gh** [comment]
Am writing this from my phone so forgive the terseness and possibly syntax errors. 

But I think we want something like the below. 

First, we want to use `with` to check that the goog analytics param is set. 

Second, we can just use the regular internal template, since it already has logic for v3 vs v4. 

See: https://github.com/gohugoio/hugo/blob/master/tpl/tplimpl/embedded/templates/google_analytics.html

```
{{ if hugo.IsProduction }}
  {{ with .Site.GoogleAnalytics }}
    {{ template "_internal/google_analytics.html" . }}
  {{ end }}
{{ end }}
```

**Comment by joaomagfreitas** [comment]
> Am writing this from my phone so forgive the terseness and possibly syntax errors.
> 
> But I think we want something like the below.
> 
> First, we want to use `with` to check that the goog analytics param is set.
> 
> Second, we can just use the regular internal template, since it already has logic for v3 vs v4.
> 
> See: https://github.com/gohugoio/hugo/blob/master/tpl/tplimpl/embedded/templates/google_analytics.html
> 
> ```
> {{ if hugo.IsProduction }}
>   {{ with .Site.GoogleAnalytics }}
>     {{ template "_internal/google_analytics.html" . }}
>   {{ end }}
> {{ end }}
> ```

Understood, looks like I added some unecessary conditions. In that case, maybe the Hugo documentation can be misunderstood as they reference this in the internal templates page

> When using Google Analytics v4 use _internal/google_analytics.html

That's why I added the condition to use `_async` for v3, but my bad as I didn't check the original source.

Will remove the conditions soon, thank you.

**Comment by joaomagfreitas** [comment]
@zwbetz-gh Ended up removing the "with .Site.GoogleAnalytics" condition, as it defined in Hugo source code:

https://github.com/gohugoio/hugo/pull/8287/files#diff-25f24f01856ed7f32bf39aafd2dcf3ebc1fb9835162d1057e2d84cb9446936c6R2

**Comment by zwbetz-gh** [comment]
> @zwbetz-gh Ended up removing the "with .Site.GoogleAnalytics" condition, as it defined in Hugo source code:
> 
> https://github.com/gohugoio/hugo/pull/8287/files#diff-25f24f01856ed7f32bf39aafd2dcf3ebc1fb9835162d1057e2d84cb9446936c6R2

Oh good catch!

**Review by zwbetz-gh — APPROVED** [review]
Teamwork! 🚀

**Pull request merged**